### PR TITLE
build using Meson instead of automake/autoconf

### PR DESCRIPTION
**Pull request by LudovicRousseau**

### CODE REVIEW

File: meson.build [hunk 1]
```diff
@@ -0,0 +1,260 @@
...
+# for generated files from .in templates
+confgen_data = configuration_data({
+  'VERSION' : meson.project_version(),
+  'ipcdir' : get_option('ipcdir'),
+  'sysconfdir' : get_option('sysconfdir'),
+  'sbindir_exp' : get_option('sbindir'),
+  'PCSCLITE_CONFIG_DIR' : get_option('serialconfdir'),
+  'usbdropdir' : get_option('usbdropdir'),
+  })
+features = []
```
Comment (Jakuje) [inline meson.build]: nit: I would move this below the compiler checks under the `# tests for options` as it is the first place accessing this array.

File: meson.build [hunk 2]
```diff
@@ -0,0 +1,260 @@
...
+  install_dir : get_option('bindir'))
+
+install_data('src/spy/setup_spy.sh',
+  install_dir : 'share/doc/pcsc-lite')
+
+run_command('pod2man',
+  ['--date=2024-01-01', 'src/spy/pcsc-spy.pod', 'pcsc-spy.1'],
+  check : true)
+install_data('pcsc-spy.1',
+  install_dir : get_option('mandir') + '/man1')
```
Comment (Jakuje) [inline meson.build]: nit: `join_paths` is more portable when merging paths than simple concatenation.

File: meson.options [hunk 3]
```diff
@@ -0,0 +1,10 @@
+option('libsystemd', type : 'boolean')
+option('libudev', type : 'boolean')
+option('libusb', type : 'boolean', value : false)
+option('polkit', type : 'boolean')
+option('usb', type : 'boolean')
+option('serial', type : 'boolean', value : false)
+option('systemdsystemunit', type : 'string', value : '/usr/lib/systemd/system')
+option('serialconfdir', type : 'string', value : '/etc/reader.conf.d')
+option('usbdropdir', type : 'string', value : '/usr/lib/pcsc/drivers')
+option('ipcdir', type : 'string', value : '/run/pcscd')
```
Comment (Jakuje) [inline meson.options]: The options support optional `description` key, which I believe would be useful for some human-readable description of the particular option.

**Review by Jakuje — COMMENTED** [review]
There are few assorted notes inline and some general in this comment.

The meson support `subdir()` option which can help to separate different parts of 250+ lines long file, avoid repeating the directories (src, doc, etc, ...) and keeping in the main one only the bare minimum. OTOH, it could make harder to navigate the buildsystem if you have to touch more files.

In any case, I am not a meson expert, but I worked with couple of projects already, so if you want some more inspiration, the https://github.com/p11-glue/p11-kit/ is using meson quite successfully for quite some time.

File: meson.build (hunk 1 above)
Comment (LudovicRousseau) [reply meson.build]: done

File: meson.build (hunk 2 above)
Comment (LudovicRousseau) [reply meson.build]: Done

File: meson.options (hunk 3 above)
Comment (LudovicRousseau) [reply meson.options]: Done

**Comment by LudovicRousseau** [comment]
The branch is now merged in https://github.com/LudovicRousseau/PCSC

**Pull request closed without merging**